<launch>

<arg name="snu_frame_id" default="SNU"/>
<arg name="occu_frame_id" default="SNU"/>
<arg name="world_frame_id" default="map"/>
<arg name="car_base_frame_id" default="/car_base_link"/>
<arg name="lidar_frame_id" default = "LidarSensor1"/>
<arg name="detected_object_frame_id" default = "map"/>

<arg name= "lane_width_nominal" default="7"/>
<arg name = "lane_csv_file" default = "$(find atypical_driving)/lane/waypoint_kiapi3.csv"/>
<arg name = "log_file_prefix" default= "$(find atypical_driving)/log/log" />
<arg name = "waypoints" default="$(find atypical_driving)/param/goal_airsim.yaml"/>

<arg name = "horizon" default = "5"/>
<arg name = "period" default = "0.1"/>
<arg name="planning_dt"     default="0.1"/> <!--?-->

<arg name ="isRearWheelDriven"    default="true"/>
<arg name = "goal/x" default = "0"/>
<arg name = "goal/y" default = "0"/>


<group ns="atypical_planning_test">
<rosparam file="$(find atypical_driving)/param/MPC_weight.yaml"/>
</group>

<arg name = "param_file"  default= "airsim.yaml"/>


<!--3. server-->
<node pkg="atypical_driving" type="integration_server" name="atypical_planning_test" output="screen">

    <param name = "local_planner/obstacle_radius_nominal" value="1"/>

    <param name="use_keti_vel" value="false"/>

    <remap from = "/occupancy_grid" to = "/costmap_node/costmap/costmap"/>

    <param name = "world_frame_id" value="$(arg world_frame_id)" />
    <param name = "snu_frame_id" value="$(arg snu_frame_id)" />
    <param name = "occu_map_frame_id" value="$(arg occu_frame_id)" />
    <param name = "base_link_id" value="$(arg car_base_frame_id)" />
    <param name = "detected_objects_id" value="$(arg detected_object_frame_id)" />

    <param name = "vmax" value = "5.0"/> <!--m/s-->


    <!--Shaping params for nominal_speed selection-->
    <param name = "vmax" value = "4.0"/> <!--m/s-->
    <param name = "vmin" value = "1.5"/> <!--m/s-->
    <param name = "v_ref_past_weight" value = "0.8"/> <!--m/s-->
    <param name = "curve_thres" value = "0.05"/> <!--m/s-->

    <param name = "goal/x" value = "$(arg goal/x)"/>
    <param name = "goal/y" value = "$(arg goal/y)"/>
    <param name = "goal_thres" value = "1.5"/>

    <param name = "lane_csv_file" value = "$(arg lane_csv_file)"/>
    <param name = "lane_width" value="$(arg lane_width_nominal)"/>
    <param name = "log_file_prefix" value = "$(arg log_file_prefix)" />

    <rosparam command="load" file="$(arg waypoints)" />

        <param name="global_planner/period" value="$(arg period)" />

    <param name="local_planner/horizon" value="$(arg horizon)" />
    <param name="local_planner/ts" value="$(arg planning_dt)" />
    <param name="local_planner/max_steer" value="0.52" />
    <param name="local_planner/max_accel" value="1" />
    <param name="local_planner/min_accel" value="-3" />
    <param name="local_planner/N_corr" value="5" />
    <param name="local_planner/isRearWheel" value="$(arg isRearWheelDriven)"/>
    <param name="local_planner/dyn_obst_range" value="30.0"/>
    <param name="local_planner/sfc_margin" value="0.4"/>
    <param name="local_planner/same_wpts_distance" value="0.01"/>
    <param name="local_planner/sameNumPointStopping" value="70"/>
    <!--smoothing_type: 0:Exponential Average, 1:Moving Average, 2:Acc(exp),Steer(mov), 3:ignore small angle -->
    <param name = "smooth_weight" value="0.7"/>
    <param name = "moving_horizon" value="20"/>
    <param name = "smoothing_type" value="2"/>
    <param name = "ignore_angle" value="0.017"/>
    <param name = "stopSpeed" value="0.5"/>

    <param name="predictor/observation_queue" value="10" />
    <param name="predictor/ref_height" value="0.4" />
    <param name="predictor/log_dir" value="$(find atypical_driving)/log/predictor" />


    <rosparam command="load" file="$(find atypical_driving)/param/$(arg param_file)"/>

</node>
</launch>
